src/zenml/integrations/airflow/orchestrators/airflow_orchestrator.py

```diff
@@ -55,6 +72,110 @@ def __init__(self, **values: Any):
+        schedule.
+
+        Args:
+            schedule: Containing the interval, start and end date and
+                a boolean flag that defines if past runs should be caught up
+                on
+        Returns:
+            Airflow configuration dict.
+        """
+        if schedule:
+            return {
+                "schedule_interval": schedule.interval_second,
+                "start_date": schedule.start_time,
+                "end_date": schedule.end_time or None,
+                "catchup": schedule.catchup or False,
+            }
+        return {
+            "schedule_interval": "@once",
+            "start_date": datetime.datetime.now() - datetime.timedelta(7),
+            "catchup": False,
+        }
+
+    def prepare_or_run_pipeline(
+        self,
+        sorted_steps: List[BaseStep],
+        pipeline: "BasePipeline",
+        pb2_pipeline: Pb2Pipeline,
+        stack: "Stack",
+        runtime_configuration: "RuntimeConfiguration",
+    ) -> Any:
+        """Create an airflow dag as the intermediate representation for the
+        pipeline. This dag will be loaded by airflow in the target environment
+        and used for orchestration of the pipeline.
+
+        How it works:
+            A new airflow_dag is instantiated with the pipeline name and among
+            others things the run schedule.
+
+            For each step of the pipeline a callable is created. This callable
+            uses the run_step() method to execute the step. The parameters of
+            this callable are pre-filled and an airflow step_operator is created
+            within the dag. The dependecies to upstream steps are then
+            configured.
+
+            Finally, the finished dag is returned.
+        """
+
+        import airflow
+        from airflow.operators import python as airflow_python
+
+        # Instantiate and configure airflow Dag with name and schedule
+        airflow_dag = airflow.DAG(
+            dag_id=pipeline.name,
+            is_paused_upon_creation=False,
+            **self._translate_schedule(runtime_configuration.schedule),
+        )
+
+        # Dictionary mapping step name to airflow_operators. This will be needed
+        # to link steps together with the operators of their upstream nodes
+        step_name_to_airflow_operator = {}
+
+        for step in sorted_steps:
+            # Create callable that will be used by airflow to execute the step
+            # within the orchestrated environment
+            def _step_callable(step_instance: "BaseStep", **kwargs):
+                # Extract run name for the kwargs that will be passed to the
+                # callable
+                run_name = kwargs["ti"].get_dagrun().run_id
+                self.run_step(
+                    step=step_instance,
+                    run_name=run_name,
+                    pb2_pipeline=pb2_pipeline,
+                )
+
+            # Create airflow step operator that contains the step callable
```

Comment:
```suggestion
            # Create airflow python operator that contains the step callable
```
Probably should avoid using step operator anywhere if possible